cobra/flux_analysis/geometric.py

```diff
@@ -42,45 +42,65 @@ def geometric_fba(model, epsilon=1E-06):
     """
 
     with model:
-        prob = model.problem
-        add_pfba(model)  # minimizes the solution space to convex hull
+        # iteration parameters
         delta = 1.0  # initialize at 1.0 to enter while loop
-        # first iteration
-        sol = model.optimize()
-        fva_sol = flux_variability_analysis(model)
-        v_n = 0.5 * abs(fva_sol['maximum'] + fva_sol['minimum'])
+        count = 2  # iteration #1 happens out of the loop
+
+        # vars and consts storage variables
         consts = []
         obj_vars = []
+        triples = []
+
+        # first iteration
+        prob = model.problem
+        add_pfba(model)  # minimizes the solution space to convex hull
+        model.optimize()
+        fva_sol = flux_variability_analysis(model)
+        mean_flux = (fva_sol["maximum"] + fva_sol["minimum"]).abs() / 2
+
         # set gFBA constraints
         for rxn in model.reactions:
             var = prob.Variable("geometric_fba_" + rxn.id,
                                 lb=0,
-                                ub=v_n[rxn.id])
+                                ub=mean_flux[rxn.id])
             upper_const = prob.Constraint(rxn.flux_expression - var,
-                                          ub=v_n[rxn.id],
+                                          ub=mean_flux[rxn.id],
                                           name="geometric_fba_upper_const_" +
                                           rxn.id)
             lower_const = prob.Constraint(rxn.flux_expression + var,
-                                          lb=fva_sol['minimum'][rxn.id],
+                                          lb=fva_sol["minimum"][rxn.id],
```

Comment:
Sorry, I meant to comment on that earlier but forgot. The better way to access a single element of a data frame is:
```python
fva_sol.at[rxn.id, "minimum"]
```

Reply:
Or you immediately loop over the data frame here since you only use the reaction ID and don't need the reaction object itself:
```python
for (rxn_id, min_flux, max_flux) in fva_sol.itertuples(index=True):
```

Reply:
I am using `rxn.flux_expression`, so I think the `Reaction` object is needed. I think it's better to go with `fva_sol.at[rxn.id, "minimum"]`.

Reply:
Oh yes.